lib/Artist.php

```diff
@@ -188,7 +246,17 @@ public static function GetOrCreate(Artist $artist): Artist{
 		}
 	}
 
+	/**
+	 * @throws Exceptions\ArtistHasArtworkException
+	 * @throws Exceptions\ArtistNotFoundException
+	 */
 	public function Delete(): void{
+		$artworkCount = count(Artwork::GetAllByArtist($this->UrlName, Enums\ArtworkFilterType::Admin, null /* submitterUserId */));
```

Comment:
We should not count like this because we're potentially instantiating a huge amount of objects just to get a count. Instead do a raw `Db::QueryBool('select exists (select ...))')` which will just return true on the first record match and then stop

Reply:
Done